src/SAWScript/Crucible/LLVM/Override.hs

```diff
@@ -688,6 +689,44 @@ refreshTerms sc ss =
 
 ------------------------------------------------------------------------
 
+-- | Generate assertions that all of the memory regions matched by an
+-- override's precondition are allocated, and meet the appropriate
+-- requirements for alignment and mutability.
+enforcePointerValidity ::
+  (?lc :: Crucible.TypeContext, Crucible.HasPtrWidth (Crucible.ArchWidth arch)) =>
+  LLVMCrucibleContext arch ->
+  W4.ProgramLoc ->
+  MS.StateSpec (LLVM arch) ->
+  OverrideMatcher (LLVM arch) md ()
+enforcePointerValidity cc loc ss =
+  do sym <- Ov.getSymInterface
+     sub <- OM (use setupValueSub) -- Map AllocIndex (LLVMPtr (Crucible.ArchWidth arch))
+     let allocs = view MS.csAllocs ss -- Map AllocIndex LLVMAllocSpec
+     let mems = Map.elems $ Map.intersectionWith (,) allocs sub
+     let w = Crucible.PtrWidth
+     let memVar = Crucible.llvmMemVar (ccLLVMContext cc)
+     mem <- readGlobal memVar
+
+     sequence_
+       [ do psz' <-
+              liftIO $ W4.bvLit sym Crucible.PtrWidth $ Crucible.bytesToInteger psz
+            c <-
+              liftIO $
+              Crucible.isAllocatedAlignedPointer sym w alignment mut ptr (Just psz') mem
+            let msg =
+                  "Pointer not valid:"
```

Comment:
It's unfortunate that we can't have a more specific diagnosis here. The tradeoff for a better API on the Crucible side seems to be that we `and` more predicates together, losing some granularity in the assertions.